src/modules/vtol_att_control/vtol_type.cpp

```diff
@@ -176,10 +177,50 @@ float VtolType::update_and_get_backtransition_pitch_sp()
 
 	const float track = atan2f(_local_pos->vy, _local_pos->vx);
 	const float accel_body_forward = cosf(track) * _local_pos->ax + sinf(track) * _local_pos->ay;
+	const float vel_body_forward = cosf(track) * _local_pos->vx + sinf(track) * _local_pos->vy;
 
-	// increase the target deceleration setpoint provided to the controller by 20%
-	// to make overshooting the transition waypoint less likely in the presence of tracking errors
-	const float dec_sp = _param_vt_b_dec_mss.get() * 1.2f;
+	const float default_dec_sp = _param_vt_b_dec_mss.get();
+	// Maximum allowed deceleration setpoint as a function of the nominal deceleration setpoint
+	const float max_dec_sp = 2.5f * default_dec_sp;
+
+	float dec_sp = default_dec_sp;
+
+	if (_attc->get_pos_sp_triplet()->current.valid && _attc->get_local_pos()->xy_valid) {
```

Comment:
Does the position setpoint triplet become invalid if you are triggering back transition from a manual flight mode? If not, why is it relevant that the validity of the current setpoint matters?